packages/devtools_app/lib/src/standalone_ui/ide_shared/property_editor/property_editor_inputs.dart

```diff
@@ -0,0 +1,223 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file or at https://developers.google.com/open-source/licenses/bsd.
+
+import 'package:devtools_app_shared/ui.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import 'property_editor_controller.dart';
+import 'property_editor_types.dart';
+
+class BooleanInput extends StatelessWidget {
+  const BooleanInput({
+    super.key,
+    required this.property,
+    required this.controller,
+  });
+
+  final FiniteValuesProperty property;
+  final PropertyEditorController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return _DropdownInput<Object>(property: property, controller: controller);
+  }
+}
+
+class DoubleInput extends StatelessWidget {
+  const DoubleInput({
+    super.key,
+    required this.property,
+    required this.controller,
+  });
+
+  final NumericProperty property;
+  final PropertyEditorController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return _TextInput<double>(property: property, controller: controller);
+  }
+}
+
+class EnumInput extends StatelessWidget {
+  const EnumInput({
+    super.key,
+    required this.property,
+    required this.controller,
+  });
+
+  final FiniteValuesProperty property;
+  final PropertyEditorController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return _DropdownInput<String>(property: property, controller: controller);
+  }
+}
+
+class IntegerInput extends StatelessWidget {
+  const IntegerInput({
+    super.key,
+    required this.property,
+    required this.controller,
+  });
+
+  final NumericProperty property;
+  final PropertyEditorController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return _TextInput<int>(property: property, controller: controller);
+  }
+}
+
+class StringInput extends StatelessWidget {
+  const StringInput({
+    super.key,
+    required this.property,
+    required this.controller,
+  });
+
+  final EditableProperty property;
+  final PropertyEditorController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return _TextInput<String>(property: property, controller: controller);
+  }
+}
+
+class _DropdownInput<T> extends StatelessWidget with _PropertyInputMixin<T> {
+  _DropdownInput({super.key, required this.property, required this.controller});
+
+  final FiniteValuesProperty property;
+  final PropertyEditorController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    return DropdownButtonFormField(
+      value: property.valueDisplay,
+      decoration: decoration(property, theme: theme),
+      isExpanded: true,
+      items:
+          property.propertyOptions.toList().map((option) {
+            return DropdownMenuItem(
+              value: option,
+              // TODO(https://github.com/flutter/devtools/issues/8531) Handle onTap.
```

Comment:
this todo links to a closed issue. Is there another issue you meant to link to? I thought this functionality was already wired up with the current property editor.